### ISSUE TITLE
task.json ignored ${fileDirname} and dump output binary into ${workspaceFolder}

### ISSUE THREAD

**Issue by aflyhorse** [microsoft/vscode-cpptools#6519]
Type: Debugger

**Describe the bug**
- OS and Version: Win10 2004, WSL Debian buster (latest)
- VS Code Version:1.51.1
- C/C++ Extension Version: v1.1.1
- Other extensions you installed (and if the issue persists after disabling them): WSL, Python, Jupyter
- A clear and concise description of what the bug is.

Since the (default) tasks.json contain ```"${fileDirname}/${fileBasenameNoExtension}"```, it should generate the output binary in ```${fileDirname}``` path instead of ```${workspaceFolder}```.

**To Reproduce**
*Please include a code sample and `launch.json` configuration.*

1. Start WSL, ```cd vscode ; code .```
2. Create a ```test.cpp``` in ```~/vscode/Cpp1/```
```#include <iostream>
int main()
{
    std::cout << "Hello!" << std::endl;
    return 0;
}
```
3. Hit F5 and select C++/GDB and then g++
4. Output ```test``` successfully built as ```~/vscode/test```
5. Error Dialog: ```launch: program '/home/username/vscode/Cpp1/test' does not exist.```

Generated tasks.json (default, unmodified)
```{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++ build active file",
            "command": "/usr/bin/g++",
            "args": [
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "/usr/bin"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Generated task by Debugger"
        }
    ],
    "version": "2.0.0"
}
```

Generated launch.json (default, unmodified)
```{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "g++ - Build and debug active file",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}/${fileBasenameNoExtension}",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ],
            "preLaunchTask": "C/C++: g++ build active file",
            "miDebuggerPath": "/usr/bin/gdb"
        }
    ]
}
```

**Additional context**
*If applicable, please include logging by adding "logging": { "engineLogging": true, "trace": true, "traceResponse": true } in your `launch.json`*
Add any other context about the problem here including log or error messages in your Debug Console or Output windows.

**Comment by sean-mcmanus** [comment]
This is a duplicate of https://github.com/microsoft/vscode-cpptools/issues/6386 -- it'll be fixed in our next release (1.2.0-insiders).

**Comment by sean-mcmanus** [comment]
Fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/1.1.2
